packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -40,18 +52,15 @@ const specificationBuilder: PermissionSpecificationBuilder<
   return {
     permissionType: PermissionType.Endowment,
     targetName: permissionName,
-    allowedCaveats: [SnapCaveatType.ChainIds],
+    allowedCaveats: [SnapCaveatType.ChainIds, SnapCaveatType.LookupMatchers],
     endowmentGetter: (_getterOptions?: EndowmentGetterParams) => undefined,
-    validator: ({ caveats }) => {
-      if (
-        !caveats ||
-        (caveats !== null && caveats?.length > 1) ||
-        (caveats?.length === 1 && caveats[0].type !== SnapCaveatType.ChainIds)
-      ) {
-        throw rpcErrors.invalidParams({
-          message: `Expected a single "${SnapCaveatType.ChainIds}" caveat.`,
-        });
-      }
+    validator: (permission) => {
```

Comment:
Just use `createGenericPermissionValidator` directly here, the return value is meant to be a valid `validator`